Ntuplizer/plugins/METsNtuplizer.cc

```diff
@@ -231,10 +237,58 @@ void METsNtuplizer::fillBranches( edm::Event const & event, const edm::EventSetu
   event.getByToken(metpuppiInputToken_, METspuppi_ );
 
   for (const pat::MET &met : *METspuppi_) {
-    nBranches_->METpuppi_et.push_back(met.et());
-    nBranches_->METpuppi_phi.push_back(met.phi());
+    nBranches_->MET_puppi_et.push_back(met.pt());
+    nBranches_->MET_puppi_phi.push_back(met.phi());
+
+  }
+
+  
+
+  // Y.T added 10 Sep. For MVA MET
+
+  event.getByToken(metmvaInputToken_, METsmva_ );
+
+  int nmva = 0;
+
+  for (const pat::MET &met : *METsmva_) {
+    nBranches_->MET_mva_et.push_back(met.pt());
+    nBranches_->MET_mva_phi.push_back(met.phi());
+    nBranches_->MET_mva_cov00.push_back(met.getSignificanceMatrix()(0,0));
+    nBranches_->MET_mva_cov01.push_back(met.getSignificanceMatrix()(0,1));
+    nBranches_->MET_mva_cov10.push_back(met.getSignificanceMatrix()(1,0));
+    nBranches_->MET_mva_cov11.push_back(met.getSignificanceMatrix()(1,1));
+
+    nmva++;
+    
+    std::vector<float> recoil_pt;
+    std::vector<float> recoil_eta;
+    std::vector<float> recoil_phi;
+    std::vector<int> recoil_pdgId;
+
+    recoil_pt.clear();
+    recoil_eta.clear();
+    recoil_phi.clear();
+    recoil_pdgId.clear();
+
+    //    std::cout << "met = " << met.pt() << std::endl;
+
+    for(auto name: met.userCandNames()){
+      reco::CandidatePtr aRecoCand = met.userCand(name);
+
+      recoil_pt.push_back(aRecoCand->p4().Pt());
+      recoil_eta.push_back(aRecoCand->p4().Eta());
+      recoil_phi.push_back(aRecoCand->p4().Phi());
+      recoil_pdgId.push_back(aRecoCand->pdgId());
+    }
+
+    nBranches_->MET_mva_recoil_pt.push_back(recoil_pt);
```

Comment:
This is for the MVA MET (we need to have this in order to check which MVA MET we should take)